app/components/category-destroy/component.js

```diff
@@ -0,0 +1,17 @@
+import Ember from 'ember';
+
+const { Component, inject: { service } } = Ember;
+
+export default Component.extend({
+  router: service(),
+  actions: {
+    destroy(category) {
+      category.deleteRecord();
+      category.save();
```

Comment:
You should do this as a callback, so only after you get confirmation from the backend. 
Together with that it's good to handle case when delete is unsuccessfull